src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -8418,7 +8420,8 @@ public override int GetHashCode()
         /// </remarks>
         internal bool IsType(Type type)
         {
-            string fullTypeName = type.FullName;
+            string? fullTypeName = type.FullName;
+            Debug.Assert(fullTypeName is not null);
```

Comment:
Maybe we should check for null?

```csharp
if (fullTypeName is null)
    return false;
```

Reply:
The `Debug.Assert` is necessary here to tell the compiler `fullTypeName` is not null.

In my opinion there should an actual null check here, but in a separate PR,

Reply:
> ```cs
> string fullTypeName = type.FullName!;
> ```

What about the possible NullReferenceException in the next line?

Reply:
Currently, the only reference to the `TypeName.IsType` method is from `FunctionMemberAst.IsReturnTypeVoid`. But the method is `internal`, it is accessible to any file in `System.Management.Automation`, so I think we should try to avoid the NullReferenceException.

Reply:
Formally fullTypeName is nullable. So I am ok with adding the Debug.Assert() here. We should avoid changing a code at annotation time.